<launch>

    <arg name="if_double" default="false"/>
    <node unless="$(arg if_double)" pkg="lslidar_x10_driver" type="lslidar_x10_driver_node" name="lslidar_x10_driver_node" output="screen">
    <param name="lidar_name" value="M10_P"/>                        #雷达选择:M10 M10_P M10_PLUS M10_GPS N10
    <param name="serial_port" value="/dev/wheeltec_lidar1"/>        #雷达连接的串口
    <param name="interface_selection" value="serial"/>              #接口选择:net 为网口,serial 为串口。
    <param name="frame_id" value="laser"/>                          #激光坐标
    <param name="min_distance" type="double" value="0"/>            #雷达接收距离最小值
    <param name="max_distance" type="double" value="100"/>          #雷达接收距离最大值
    <param name="scan_topic" value="scan"/>                         #设置激光数据topic名称
    <param name="use_gps_ts" value="false"/>                        #雷达是否使用GPS授时
    <param name="truncated_mode" value="0"/>                        <!--0:雷达不屏蔽角度 1:雷达屏蔽角度-->
    <rosparam param="disable_min">[130]</rosparam>                  <!--雷达屏蔽角度左值（角度顺时针方向，支持多角度屏蔽）-->
    <rosparam param="disable_max">[230]</rosparam>                  <!--雷达屏蔽角度右值（角度顺时针方向，支持多角度屏蔽）-->
    </node>

  <node if="$(arg if_double)" pkg="lslidar_x10_driver" type="lslidar_x10_driver_node" name="M10_P_1" output="screen">
    <param name="lidar_name" value="M10_P"/>                        #雷达选择:M10 M10_P M10_PLUS M10_GPS N10
    <param name="serial_port" value="/dev/wheeltec_lidar1"/>        #雷达连接的串口
    <param name="interface_selection" value="serial"/>              #接口选择:net 为网口,serial 为串口。
    <param name="frame_id" value="laser1"/>                         #激光坐标
    <param name="min_distance" type="double" value="0"/>            #雷达接收距离最小值
    <param name="max_distance" type="double" value="100"/>          #雷达接收距离最大值
    <param name="scan_topic" value="scan1"/>                        #设置激光数据topic名称
    <param name="use_gps_ts" value="false"/>                        #雷达是否使用GPS授时
  </node>

  <node if="$(arg if_double)" pkg="lslidar_x10_driver" type="lslidar_x10_driver_node" name="M10_P_2" output="screen">
    <param name="lidar_name" value="M10_P"/>                        #雷达选择:M10 M10_P M10_PLUS M10_GPS N10
    <param name="serial_port" value="/dev/wheeltec_lidar2"/>        #雷达连接的串口
    <param name="interface_selection" value="serial"/>              #接口选择:net 为网口,serial 为串口。
    <param name="frame_id" value="laser2"/>                         #激光坐标
    <param name="min_distance" type="double" value="0"/>            #雷达接收距离最小值
    <param name="max_distance" type="double" value="100"/>          #雷达接收距离最大值
    <param name="scan_topic" value="scan2"/>                        #设置激光数据topic名称
    <param name="use_gps_ts" value="false"/>                        #雷达是否使用GPS授时
    
  </node>

     <node if="$(arg if_double)" name="laser_double" pkg="laser_double" type="laser_double" output="screen">
        <param name="frame_id"         type="string"   value="laser"/> <!--雷达输出frame_id-->
        <param name="scan_topic"         type="string"   value="scan"/> <!--雷达输出话题-->
        <param name="truncated_mode" value="1"/>    <!--0:两雷达不屏蔽角度 1:两雷达屏蔽角度-->

        <param name="scan1_topic"         type="string"   value="scan1"/>   <!-- 雷达1话题-->
        <param name="lidar1_rotate"    type="double"   value="-135"/>       <!-- 雷达1坐标旋转角度,小车前方为0度角，正方向为顺时针-->
        <param name="lidar1_xoffset"   type="double"   value="0.3"/>       <!-- 雷达1坐标原点x轴偏移距离-->
        <param name="lidar1_yoffset"   type="double"   value="0.235"/>    <!-- 雷达1坐标原点y轴偏移距离-->
        <rosparam param="disable_min1">[120]</rosparam>                      <!--雷达1屏蔽角度左值（角度顺时针方向，支持多角度屏蔽）-->
        <rosparam param="disable_max1">[240]</rosparam>                     <!--雷达1屏蔽角度右值（角度顺时针方向，支持多角度屏蔽）-->

        <param name="scan2_topic"         type="string"   value="scan2"/>  <!-- 雷达2话题-->
        <param name="lidar2_rotate"    type="double"   value="45"/>       <!-- 雷达2坐标旋转角度,小车前方为0度角，正方向为顺时针-->
        <param name="lidar2_xoffset"   type="double"   value="-0.3"/>      <!-- 雷达2坐标原点x轴偏移距离-->
        <param name="lidar2_yoffset"   type="double"   value="-0.235"/>     <!-- 雷达2坐标原点y轴偏移距离-->
        <rosparam param="disable_min2">[120]</rosparam> <!--雷达2屏蔽角度左值（角度顺时针方向，支持多角度屏蔽）-->
        <rosparam param="disable_max2">[240]</rosparam> <!--雷达2屏蔽角度右值（角度顺时针方向，支持多角度屏蔽）-->
    </node>

</launch>
